stdlib/internal/promql/join.go

```diff
@@ -464,7 +464,7 @@ func appendRowToBuilder(builder *execute.ColListTableBuilder, obj values.Object)
 	obj.Range(func(name string, v values.Value) {
 		idx := execute.ColIdx(name, builder.Cols())
 		if idx < 0 {
-			err = fmt.Errorf("column %s not found", name)
+			err = errors.Newf(codes.NotFound, "column %s not found", name)
```

Comment:
I've been using `FailedPrecondition` for it. It implies a user error, but it indicates that the data may be the cause of the problem because the section is dynamic which seemed to fit into `FailedPrecondition`. InfluxDB translates both of them to `Invalid` though so it doesn't matter much, but it would lend consistency with my other PR.

Reply:
I didn't know this would result in a 404. I'll change the codes to internal.